crates/sui-framework/packages/dwallet-system/sources/dwallet.move

```diff
@@ -287,21 +298,70 @@ module dwallet_system::dwallet {
         transfer::freeze_object(sign_session);
     }
 
+    #[allow(unused_field)]
+    struct MaliciousAggregatorSignOutput has key {
+        id: UID,
+        aggregaor_pk: vector<u8>,
+        epoch: u64,
+        signatures: vector<vector<u8>>,
+        messages: vector<vector<u8>>,
+        dwallet_id: ID,
+    }
+
+    struct MaliciousAggregatorEvent has copy, drop {
+        aggregaor_pk: vector<u8>,
+        epoch: u64,
+        signatures: vector<vector<u8>>,
+        messages: vector<vector<u8>>,
+        dwallet_id: ID,
+    }
+
+
     #[allow(unused_function)]
     /// This function is called by blockchain itself.
     /// Validtors call it, it's part of the blockchain logic.
     /// NOT a native function.
-    fun create_sign_output<S: store>(session: &SignSession<S>, signatures: vector<vector<u8>>, ctx: &mut TxContext) {
+    fun create_sign_output<S: store>(
```

Comment:
instead of storing the hash in the sign session - try to move validation logic to the secp module, and call the functions that create the objects from there